src/weights.jl

```diff
@@ -25,6 +25,15 @@ isempty(wv::AbstractWeights) = isempty(wv.values)
 Base.getindex(wv::AbstractWeights, i) = getindex(wv.values, i)
 Base.size(wv::AbstractWeights) = size(wv.values)
 
+function Base.setindex!(wv::AbstractWeights{S, T}, v::T, i::Int) where {S, T}
```

Comment:
`v` could be any `Real`, no need to restrict it. But then to avoid type instabilities better do `wv.sum = wv.sum - wv[i] + v`.

Would also be good to use `@propagate_inbounds`.